src/native/libs/System.Native/pal_io.c

```diff
@@ -1152,9 +1173,46 @@ static int32_t CopyFile_ReadWrite(int inFd, int outFd)
 }
 #endif // !HAVE_FCOPYFILE
 
+
+#ifdef __NR_copy_file_range
+static ssize_t CopyFileRange(int inFd, int outFd, size_t len)
+{
+    return syscall(__NR_copy_file_range, inFd, NULL, outFd, NULL, len, 0);
+}
+
+static bool SupportsCopyFileRange()
+{
+    static volatile int s_isSupported = 0;
+
+    int isSupported = s_isSupported;
+    if (isSupported == 0)
+    {
+        isSupported = -1;
+
+        // Avoid known issues with copy_file_range that are fixed in Linux 5.3+ (https://lwn.net/Articles/789527/).
+        struct utsname name;
+        if (uname(&name) == 0)
+        {
+            if (name.release[1] != '.' ||
```

Comment:
Can name.release be an empty string in theory? If yes, this should be `name.release[0] != 0 && name.release[1] != '.'` and similar for the other `.` check.

Reply:
I don't think they are needed in practice, but I've added them anyway.